sdk/bare/common/usr/demo/task_cc.h

```diff
@@ -18,5 +18,6 @@ void task_cc_pwm(int cc_idx, uint8_t phase, uint8_t pwm_chnl);
 void task_cc_adc(int cc_idx, uint8_t phase, uint8_t adc_chnl, double adc_gain, double adc_offset);
 void task_cc_tune(int cc_idx, double Rs, double Ld, double Lq, double bw);
 void task_cc_set(int cc_idx, double Id_star, double Iq_star, double omega_e);
+void task_cc_set_w_theta(int cc_idx, double Id_star, double Iq_star, double theta_e);
```

Comment:
I think approach would work (i.e. setting with a theta), but it might be clearer if you create different explicit "modes" for the cc module. Your code does this, but implicitly. Consider doing the following:

1. Create a mode variable in each cc context which stores the operating mode (maybe called `mode` as an enum type).
2. Create a function which let's the user set the mode.
3. Change the "set" functions to check the current mode of the cc context and only do their actions if it's in the right mode. For example, the `task_cc_set_w_theta` function is only valid in the theta-controlled mode, so if it's called in the omega-controlled mode, it should return without doing anything.

Reply:
I think I have made this change in commit '09777e1', please let me know if I missed something.